src/main/java/core/basesyntax/MyLinkedList.java

```diff
@@ -3,48 +3,145 @@
+            Node<T> node = new Node<>(null, value, firsNode);
+            firsNode.prev = node;
+            firsNode = node;
+            size++;
+            return;
+        }
+        Node<T> node = getNode(index);
+        Node<T> newNode = new Node<>(node.prev, value, node);
+        node.prev.next = newNode;
+        node.prev = newNode;
+        size++;
     }
 
     @Override
     public boolean addAll(List<T> list) {
-        return false;
+        for (int i = 0; i < list.size(); i++) {
+            add(list.get(i));
+        }
+        return true;
     }
 
     @Override
     public T get(int index) {
-        return null;
+        if (index < 0 || index > size) {
+            throw new IndexOutOfBoundsException();
+        }
+        return getNode(index).element;
     }
 
     @Override
     public T set(T value, int index) {
-        return null;
+        Node<T> x = getNode(index);
+        T oldValue = x.element;
+        x.element = value;
+        return oldValue;
     }
 
     @Override
     public T remove(int index) {
-        return null;
+        Node<T> node = getNode(index);
+        if (size == 1) {
+            lastNode = null;
+            firsNode = null;
+            size--;
+            return node.element;
+        }
+        if (index == 0) {
+            node.next.prev = null;
+            firsNode = node.next;
+            size--;
+            return node.element;
+        }
+        if (index == size - 1) {
+            node.prev.next = null;
+            lastNode = node.next;
+            size--;
+            return node.element;
+        }
+        node.prev.next = node.next;
+        node.next.prev = node.prev;
+        size--;
+        return node.element;
     }
 
     @Override
     public boolean remove(T t) {
+        for (int i = 0; i < size; i++) {
+            if (t == getNode(i).element || getNode(i).element.equals(t)) {
+                remove(i);
+                return true;
+            }
+        }
         return false;
     }
 
     @Override
     public int size() {
-        return 0;
+        return size;
     }
 
     @Override
     public boolean isEmpty() {
-        return false;
+        return size == 0;
+    }
+
+    private static class Node<T> {
+        T element;
+        Node<T> next;
+        Node<T> prev;
+
+        Node(Node<T> prev, T element, Node<T> next) {
+            this.element = element;
+            this.prev = prev;
+            this.next = next;
+
+        }
+    }
+
+    // присоединяет к последней node
+    public void addToLast(T value) {
+        if (isEmpty()) {
+            Node<T> newNode = new Node<>(null, value, null);
+            firsNode = lastNode = newNode;
+        } else {
+            Node<T> newNode = new Node<>(lastNode, value, null);
+            lastNode.next = newNode;
+            lastNode = newNode;
+        }
+        size++;
+    }
+
+    // определяется узел, который находиться под нужным индексом
+    Node<T> getNode(int index) {
```

Comment:
- in this method let's start moving from the first Entry OR from the last one
depending on whether index is closer to the start or to the end’
- add access modifier